src/core/actions.ts

```diff
@@ -45,6 +46,30 @@ namespace Actions {
   export type UpdatePageSize = Action<typeof UPDATE_PAGE_SIZE, number>;
   export const UPDATE_CURRENT_PAGE = 'UPDATE_CURRENT_PAGE';
   export type UpdateCurrentPage = Action<typeof UPDATE_CURRENT_PAGE, number>;
+  export const UPDATE_QUERY = 'UPDATE_QUERY';
+  export type UpdateQuery = Action<typeof UPDATE_QUERY, string>;
+  export const RESET_REFINEMENTS = 'RESET_REFINEMENTS';
+  export type ResetRefinements = Action<typeof RESET_REFINEMENTS, boolean | string>;
+  export const RESET_PAGE = 'RESET_PAGE';
+  export type ResetPage = Action<typeof RESET_PAGE, undefined>;
+  export const ADD_REFINEMENT = 'ADD_REFINEMENT';
+  export type AddRefinement = Action<typeof ADD_REFINEMENT, Actions.Payload.Navigation.AddRefinement>;
+
+  // batch actions
+  // tslint:disable-next-line max-line-length
+  export type SwitchRefinement = [Actions.ResetPage, Actions.ResetPage, Actions.ResetRefinements, Actions.ResetPage, Actions.AddRefinement];
+  export type Search = [Actions.ResetPage, Actions.ResetPage, Actions.ResetRefinements, Actions.ResetPage, Actions.AddRefinement];
+  // tslint:disable-next-line max-line-length
+  export type ResetRecall = [Actions.ResetPage, Actions.ResetPage, Actions.ResetRefinements, Actions.ResetPage, Actions.UpdateQuery] |
+    [Actions.ResetPage, Actions.ResetPage, Actions.ResetRefinements, Actions.ResetPage, Actions.UpdateQuery, Actions.ResetPage, Actions.SelectRefinement];
+    // tslint:disable-next-line max-line-length
+  // export type UpdateSearch = Array<Actions.ResetPage | Actions.UpdateQuery | Actions.ResetRefinements | Actions.SelectRefinement | Actions.AddRefinement>;
```

Comment:
we should uncomment this type and remove the other `type UpdateSearch` defined in this file